File: update_links.js
const fs = require('fs');
const path = require('path');

const baseDir = path.resolve('c:/xampp/htdocs/fnff');
let count = 0;

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (!fullPath.includes('.gemini') && !fullPath.includes('node_modules') && !fullPath.includes('tmp') && !fullPath.includes('.git')) {
                walkDir(fullPath);
            }
        } else if (file === 'index.html') {
            const relPath = path.relative(baseDir, dir);
            if (relPath === 'donate-us') continue;

            const depth = relPath === '' ? 0 : relPath.split(path.sep).length;
            const donateRel = depth === 0 ? 'donate-us/' : '../'.repeat(depth) + 'donate-us/';

            let content = fs.readFileSync(fullPath, 'utf8');
            let newContent = content.replace(/<a href="https:\/\/wa\.link\/[^"]*"([^>]*)>([\s\S]*?Donate Now[\s\S]*?)<\/a>/gi, `<a href="${donateRel}"$1>$2</a>`);
            newContent = newContent.replace(/<a href="(?:(?:\.\.\/)*)?contact(?:\/)?"([^>]*)>([\s\S]*?Donate Now[\s\S]*?)<\/a>/gi, `<a href="${donateRel}"$1>$2</a>`);

            if (content !== newContent) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated: ${relPath}`);
                count++;
            }
        }
    }
}

walkDir(baseDir);
console.log(`Total files updated: ${count}`);

File: update_links.py
import os
import re

base_dir = r"c:\xampp\htdocs\fnff"
count = 0

for root, dirs, files in os.walk(base_dir):
    if ".gemini" in root or "node_modules" in root or "tmp" in root or ".git" in root:
        continue
    for file in files:
        if file == "index.html":
            file_path = os.path.join(root, file)
            
            rel_path = os.path.relpath(root, base_dir)
            if rel_path == ".":
                depth = 0
            else:
                depth = len(rel_path.split(os.sep))
                
            # skip the donate-us actual file
            if rel_path == "donate-us":
                continue
                
            if depth == 0:
                donate_rel = "donate-us/"
            else:
                donate_rel = "../" * depth + "donate-us/"
                
            try:
                with open(file_path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Replace wa.link links containing Donate Now
                new_content = re.sub(
                    r'<a href="https://wa\.link/[^"]*"([^>]*)>([\s\S]*?Donate Now[\s\S]*?)</a>',
                    rf'<a href="{donate_rel}"\1>\2</a>',
                    content,
                    flags=re.IGNORECASE
                )
                
                new_content = re.sub(
                    r'<a href="(?:(?:\.\./)*)?contact(?:/)?"([^>]*)>([\s\S]*?Donate Now[\s\S]*?)</a>',
                    rf'<a href="{donate_rel}"\1>\2</a>',
                    new_content,
                    flags=re.IGNORECASE
                )
                
                if content != new_content:
                    with open(file_path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated: {rel_path}")
                    count += 1
            except Exception as e:
                print(f"Error reading {file_path}: {e}")

print(f"Total files updated: {count}")
